src/bijectors/radial_layer.jl

```diff
@@ -13,78 +13,85 @@ using Roots # for inverse
 # RadialLayer #
 ###############
 
-# FIXME: using `TrackedArray` for the parameters, we end up with
-# nested tracked structures; don't want this.
-mutable struct RadialLayer{T1,T2} <: Bijector{1}
+mutable struct RadialLayer{T1 <: Real, T2 <: AbstractVector{<:Real}} <: Bijector{1}
     α_::T1
     β::T1
     z_0::T2
 end
 
 function RadialLayer(dims::Int, container=Array)
-    α_ = container(randn(1))
-    β = container(randn(1))
-    z_0 = container(randn(dims, 1))
+    α_ = randn()
+    β = randn()
+    z_0 = container(randn(dims))
     return RadialLayer(α_, β, z_0)
 end
 
-h(α, r) = 1 / (α + r)    # for radial flow from eq(14)
-dh(α, r) = - h(α, r) ^ 2 # for radial flow; derivative of h()
+h(α, r) = 1 ./ (α .+ r)     # for radial flow from eq(14)
+#dh(α, r) = .- (1 ./ (α .+ r)) .^ 2   # for radial flow; derivative of h()
 
 # An internal version of transform that returns intermediate variables
-function _transform(flow::RadialLayer, z)
-    # from A.2
-    α = softplus.(flow.α_)
-
-    # from A.2
-    β_hat = @. -α + softplus(flow.β)
-
-    r = sqrt.(sum((z .- flow.z_0).^2; dims = 1))
-    transformed = @. z + β_hat * h(α, r) * (z - flow.z_0)   # from eq(14)
-    return (transformed=transformed, α=α, β_hat=β_hat, r=r)
+function _transform(flow::RadialLayer, z::AbstractVecOrMat)
```

Comment:
Can just remove this definition and replace the uses of `_transform` with `_radial_transform`, no?

Reply:
Sure but this is more organized. I don't want to extract fields from `flow` in too many places.